### ISSUE TITLE
backend: create endpoint for clothes

### ISSUE THREAD

**Issue by siwonpawel** [JhosefCardich/online-shop#34]
- [ ] add clothing
- [ ] edit clothing
- [ ] remove clothing
- [ ] find clothing by category
- [ ] find clothing by parent category (containing all child categories)

question: what will happend with invoices when user will delete clothing, should we consider marking them as non-active or what?

**Comment by APEX-JHOSEF-CARDICH-PALMA** [comment]
, we have to organize the tasks for each type of user. Depending on what type of user is active, the user can do certain operations or not.
In this endpoint
I think these two operations have to be available for the customer and  the  employee 

- [ ] find clothing by category

- [ ] find clothing by parent category (containing all child categories)

---
these operations just for the employe: 
- [ ] add clothing
- [ ] edit clothing
- [ ] remove clothing

***basically a customer just only can browse trough categories clothes, buy products and update its shopping cart, but can't change anything or update info on the app (There's an exception: When the user buys something, an endpoint or functionality has to decrement the number of products units in the DB )***


What do you think @siwonpawel? 👍

**Comment by siwonpawel** [comment]
Great idea, the basic authentication is implemented now so It will require enchantment
